contents/blog/feature-flag-best-practices.md

```diff
@@ -0,0 +1,198 @@
+
+To bootstrap your flags, add the relevant distinct ID and feature flag data when initializing. In Javascript, this looks like this:
+
+```js
+posthog.init('<ph_project_api_key>', {
+    api_host: '<ph_instance_address>',
+    bootstrap: {
+        distinctID: 'your-anonymous-id',
+        featureFlags: {
+            'flag-1': true,
+            'variant-flag': 'control',
+            'other-flag': false
+        }
+    }
+})
+```
+
+You can get values for the bootstrap object by using a server-side library. Call `getAllFlags` with a server-side library, then add those values as the `featureFlags` object in your client-side initialization. This enables your flags to be available instantly on page load. See more about bootstrapping flags in [our JavaScript docs](/docs/integrate/client/js#bootstrapping-flags).
+
+> Bootstrapping your flags also ensures events have accurate feature flag data. If you capture events before receiving feature flags data, data can be missing. Bootstrapping prevents this.
+
+## 3. Fallback to working code
+
+If a feature flag returns `false` or fails and returns `none`, you want to make sure you fallback to working code. This is a best practice for a few reasons:
+
+- If there is a PostHog-related problem with the flag, we return `None`. Having a fallback handles this.
+
+- If you use experiments, the fallback is the control the test is comparing against. If your fallback doesn’t work properly, it causes the experiment to be inaccurate.
+
+- If you rollback your code using a feature flag, knowing you are rolling back to working code helps you do it quickly.
+
+To fallback consistently, keep new changes separate from existing code and behind a feature flag. Test that flags returning `false` fallback to old code and that the fallback code works. This ensures your apps continue to work, even if the code behind the flag isn’t used.
+
+## 4. Accurately identify users
+
+Because feature flags evaluate based on the distinct ID of the user, having accurate identification is critical. Not knowing your user limits the flags you can show to them. If you identify users accurately, you can better target your feature flags to them.
+
+Having at least a “sticky” ID (like a cookie, which is the default) ensures the user gets a consistent flag evaluation in your product. Without identification, PostHog wouldn’t know what feature flags to show them. We default to not showing flags, but if the distinct ID changes often, we could show multiple variants to the same user.
+
+Accurate identification includes setting up group analytics and person properties if you use them to rollout feature flags. Users must be a part of groups or have a property before PostHog can decide to show them a feature flag relying on that group or property. 
+
+## 5. Name your feature flags well
+
+Here is some advice on naming your feature flags, they should:
```

Comment:
```suggestion
Here is some practical advice on naming your feature flags to avoid confusion. None of these are laws and you can create your own conventions, but these are the ones we follow:
```